source/runtime.cpp

```diff
@@ -4882,21 +4892,49 @@ static std::string expand_macro_string(const std::string &input, std::vector<std
-
-	std::string screenshot_name = _screenshot_name;
-	screenshot_name += postfix;
-	screenshot_name += (_screenshot_format == 0 ? ".bmp" : _screenshot_format == 1 ? ".png" : ".jpg");
-
-	const std::filesystem::path screenshot_path =
-		g_reshade_base_path / std::filesystem::u8path(
-			expand_macro_string((_screenshot_path / std::filesystem::u8path(screenshot_name)).u8string(), {
-				{ "AppName", g_target_executable_path.stem().u8string() },
+/* previous implementation with both path and name being expanded benefitted from a local vector to avoid
+    redundancy. I also had it instantiate without Count which was emplaced back after expanding the path string
+	to prevent the possibility of adding Count to your path, thus creating a new directory per screenshot
+    Technically possible again. Should we add a check to ignore Count if there are any directory separators following it?
+	Or do we trust that the user is smart enough not to do that? Probably we just don't care
+ 	So this can be eliminated along with some of the other local variables in all likelihood
+*/
+  std::vector<std::pair<std::string, std::string>> macros = {
+      {"BeforeAfter", std::string(postfix)},
+      {"AppName", g_target_executable_path.stem().u8string()},
 #if RESHADE_FX
-				{ "PresetName",  _current_preset_path.stem().u8string() },
+      {"PresetName", _current_preset_path.stem().u8string()},
+      {"Count", std::to_string(screenshot_count)}
 #endif
-				{ "Count", std::to_string(screenshot_count) }
-			}));
+  };
+
+ // Could likely consolidate this with some inlining
+ std::string expanded_screenshot_name = expand_macro_string(_screenshot_name, macros);
+	
+ // Trim leading and trailing whitespaces from the name.
+ // Was more of an issue with old split implementation so this could stay or go
+ expanded_screenshot_name.erase(0, expanded_screenshot_name.find_first_not_of(" \t\n\r\f\v"));
+ expanded_screenshot_name.erase(expanded_screenshot_name.find_last_not_of(" \t\n\r\f\v") + 1);
+
+
+  // If save before/after option is checked but not used as a macro, append it to the filename (old behavior)
+  bool before_after_used = (expanded_screenshot_name.find(std::string(postfix)) != std::string::npos);
+  // If we want to fix tiny issue of overlay being included we'd do another check here I suppose
+  if (_screenshot_save_before && !before_after_used) {
+    expanded_screenshot_name += "_" + std::string(postfix);
+  }
+
+  expanded_screenshot_name += (_screenshot_format == 0 ? ".bmp" : _screenshot_format == 1 ? ".png" : ".jpg");
+
+  // Normalize path by removing duplicate slashes, handling mixed slashes, again likely unnecessary use of local var
+  std::string normalized_name = expanded_screenshot_name;
+  std::replace(normalized_name.begin(), normalized_name.end(), '/', '\\');
+  while (normalized_name.find("\\\\") != std::string::npos) {
+    normalized_name.replace(normalized_name.find("\\\\"), 2, "\\");
+  }
```

Comment:
This can be removed by taking advantage of std::filesystem::path functionality: Can just do `const std::filesystem::path screenshot_path = g_reshade_base_path / std::filesystem::u8path(normalized_name).lexically_normal()` below.